Cleanup old methods

jbrichau opened the pull request:
- Removes old code mentioned in issue #1417
- Removes GRPackage extensions. This mechanism for declaring dependencies between packages pre-dates the Metacello configuration and baseline. It's no longer up to date and unused.
- Remove broken `WADevelopment>>#cleanupCategories` (see #1343). This can now be better handled via code critiques as well.

marschall commented:
Maybe I'm missing something here. Unfortunately we still have some senders of `GRPackage class >> #grPackages`

- `JQUiFunctionalTest >> #packages`
- `JQFunctionalTest >> #seasidePackagesOn:`
- `JQFunctionalTest >> #seasidePackagesOn:`
- `WADevelopment class >> # graphFunctionalDependencies:tests:`
- `WADevelopment class >> #allGRPackages`
  - `WADevelopment class >> #allDirtyPackages`
  - `WADevelopment class >> #browserEnvironment`
  - `WADevelopment class >> #cleanupCategories`
  - `WADevelopment class >> #gofers`
  - `WADevelopment class >> #graphFunctionalDependencies:tests:`

jbrichau commented:
Yes, that was not so good... just did not check those.

- In the functional tests, I replaced its use with something completely different (country names and currencies) to ensure these are not dependent on a specific platform.
- In WADevelopment, I removed the methods that referred it. None of these have been in use for the last 10 years and are replaced by either Metacello or code critiques in Pharo.

marschall reviewed (approved):
LGTM
